app/Project.swift

```diff
@@ -0,0 +1,72 @@
+import ProjectDescription
+
+func tuistAppDependencies() -> [TargetDependency] {
+    [
+        .external(name: "Path"),
+        .external(name: "TuistSupport"),
+        .external(name: "TuistCore"),
+        .external(name: "TuistServer"),
+        .external(name: "TuistAutomation"),
+        .external(name: "Mockable"),
+        .external(name: "XcodeGraph"),
+    ]
+}
+
+let project = Project(
+    name: "Tuist",
+    settings: .settings(
+        debug: [
+            "SWIFT_ACTIVE_COMPILATION_CONDITIONS": "$(inherited) MOCKING",
+        ]
+    ),
+    targets: [
+        .target(
+            name: "Tuist",
+            destinations: .macOS,
+            product: .app,
+            bundleId: "io.tuist.Tuist.App",
+            deploymentTargets: .macOS("14.0.0"),
+            infoPlist: .extendingDefault(
+                with: [
+                    "CFBundleURLTypes": .array(
+                        [
+                            Plist.Value.dictionary(
+                                [
+                                    "CFBundleTypeRole": "Viewer",
+                                    "CFBundleURLName": "io.tuist.Tuist.App",
+                                    "CFBundleURLSchemes": .array(["tuist"]),
+                                ]
+                            ),
+                        ]
+                    ),
+                    "LSUIElement": .boolean(true),
+                ]
+            ),
+            sources: ["TuistApp/Sources/**"],
+            resources: ["TuistApp/Resources/**"],
+            dependencies: tuistAppDependencies(),
+            settings: .settings(
+                base: [
+                    "DEVELOPMENT_TEAM": "U6LC622NKF",
+                    "CODE_SIGN_STYLE": "Automatic",
+                    "CODE_SIGN_IDENTITY": "Apple Development",
+                ]
+            )
+        ),
+        .target(
+            name: "TuistAppTests",
+            destinations: .macOS,
+            product: .unitTests,
+            bundleId: "io.tuist.TuistAppTests",
```

Comment:
I'd rather keep the current convention of `io.tuist.{name-of-target}`. I did change the bundle id of the app target but only because `io.tuist.Tuist` was not available (and it looks a bit off in this case)